### DISCUSSION TITLE
sx126x Calibration Image  need timeout?

### DISCUSSION THREAD

**Discussion by alexrayne** [Lora-net/LoRaMac-node#1008] [Q&A]
Hallow!
i tests sample modem application for SX1261 transiver chip.

And found that when after command RADIO_CALIBRATEIMAGE goes TX Freq setup - it fails somehow. When i make pause more then 0.7ms it work well.

Can anybody explain behaviour of calibration commands? It not produces BUSY signal, but need timeout?

**Comment by mluis1** [comment]
The description you provide is not very clear and it is hard to understand what you are the steps being executed.

Could you tell us the steps to reproduce the issue?
Even better can you provide an example code which reproduces the issue?

In the future it would be nice if you could post this kind of questions on the project [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab. It is a better place to engage discussions and then we can agree if it is an issue or not.
If you don't mind I move this issue to the [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab.

**Comment by alexrayne** [comment]
lets say it again.
my sx16x-board comunicate with sx1261 chip via SPI, now right according datasht demands - it waits that BUSY inactive before any SPI commands.
now my situation: 
1) i have at chip statup initialisation sequence that contains - right after CalibrationImage command, my stack pass setup Frequency command, and later sequens leads to transmit command .
2) this initialisation leads to successful transmit if infered additional pause between above commands more then 800us (i infer 1ms now) 
   If infer less pause, no transmition resulted. (Maybe chip make transmition, but on some frequency, that my gateway not accept.)

in datasheet on sx126x sayd that whole Calibration takes about 3.5ms. It not say how much takes Calibration Image. And looks that any Calibration not locks chip at BUSY.
But, as described before, without pause after calibration, my transmition fails.

Can anybody explain - where is a bug? This lora-stack miss this pause, or some another reason of failure i should search?

**Reply by mluis1** [reply]
I'm sorry, but, your description is still not clear and hard to understand the observed issue.

It would maybe be better to provide an example code which reproduces your observed issue.

Using the code provided on this repository does not show issues related to the image calibration operation.

I would recommend to post your question on below forum in order to get better support on Semtech radios usage.

https://forum.lora-developers.semtech.com/

**Reply by alexrayne** [reply]
i solve my problem, as i say above. so i not interested in discussion any more.
To descrive more: in sx1261 driver, in ImageCalibration function i infer delay 1ms right after RADIO_CALIBRATEIMAGE sent to chip.
With this delay send work relyable. 

Standart sequence for sending: `power on -> calibrate -> setup freq -> send`
Driver reports that data sent anyway, but without delay, i frequentry loose sent packets. Possibly it loose, cause frequency setup  or send command too early, and breaks calibration.

**Reply by mluis1** [reply]
Nice to hear that you solved your issue.

It is just that an example code on how to reproduce the issue would be interesting in order to understand the root cause of your observation.

Any way thank you for your feedback.

**Reply by alexrayne** [reply]
the bug spotes right at JOIN connection. 
I used renesas implementation of your stack. It provide a lot of debug printing, (and sx1261 SPI driver can prints its activity). they provide `LoRaMacDebug.c` for this printing.
So, while lora debug prints it`s messages, i have join ok. When i disable debuging - join fails ordinary, success very rare. 
Cause of this i explained above - some prints after RADIO_CALIBRATEIMAGE helps radio.
I can provide a dump of SPI activity, but need a few month to access lora-kit

**Reply by mluis1** [reply]
OK thanks, 

Then it is most certainly an issue on how this project has been ported to the Renessas MCU as we don't observe the same issue on this project supported platforms.

I would recommend that you contact the Renessas port maintainer in order to get a better support on this issue.
Or you can post a question on the [Semtech forum](https://forum.lora-developers.semtech.com/) indicating which MCU you are using and by putting a link to this discussion for reference.

**Reply by alexrayne** [reply]
i`ve alredy inspects renesas stack about this, Luis. Renesas code a bit better, then your, but difference on sx1261 drive not so big. And code that affected by my problem is practicaly identical.

You can close this issue freely, i not insit on your help. I don know how move it to discussion tab, that you propose. If you can - do it, please?

But bug keep stay on semtech side shurely: since documentation on chip says nothing about - what happens if you start frequency setup during calibration. Or anybody have answer?
